lib/tests/misc.nix

```diff
@@ -88,6 +87,35 @@ runTests {
     expected = [ "2001" "db8" "0" "0042" "" "8a2e" "370" "" ];
   };
 
+  testIsStorePath =  {
+    expr =
+      let goodPath =
+            "/nix/store/d945ibfx9x185xf04b890y4f9g3cbb63-python-2.7.11";
+      in {
+        storePath = isStorePath goodPath;
+        storePathAppendix = isStorePath
+          "${goodPath}/bin/python";
```

Comment:
I wonder. We could change the semantics of the function. @edolstra would that make sense?
In any case, it’s nice that this test case documents the current semantics.